src/libraries/System.Private.CoreLib/src/System/IO/FileStreamOptions.cs

```diff
@@ -109,5 +112,31 @@ public int BufferSize
             get => _bufferSize;
             set => _bufferSize = value >= 0 ? value : throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_NeedNonNegNum);
         }
+
+        /// <summary>
+        /// Unix file mode used when a new file is created.
+        /// </summary>
+        public UnixFileMode? UnixCreateMode
```

Comment:
Currently I've added the `UnsupportedOSPlatform` to the setter only. This allows me to use get `UnixCreateMode` in code that is also used in Windows.
Should I apply it also to the getter and make that throw?
That will require some changes to the `FileStreamStrategy` related classes which pass this value around.

Reply:
> Should I apply it also to the getter and make that throw?

In general, we don't like property getters that throw. Returning `null` on Windows makes sense to me.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property

> ❌ AVOID throwing exceptions from property getters.

Reply:
Based on this guideline I've also changed `FileSystemInfo.UnixFileMode` to no longer throw PNSE on Windows.